discussion/fairness/fairness-on-controller-race.md

```diff
@@ -0,0 +1,59 @@
+When verifying interactions between controllers, we find it hard to specify the fairness assumptions that allow controllers to make progress while there are races on the resource version number of the same object (stored in etcd).
+
+## Background
+Kubernetes controllers often employ a pattern that it reads an object from etcd (which uses version control for stored objects) and then writes the updated object back to etcd.
```

Comment:
"version control" is easily confused with something like git here, maybe call it version-based concurrency control. You can also mention the version numbers behave like logical timestamps.